### PR TITLE
[AIT-1138] Deprecate RTPO8d: `PathObject#instance` wraps primitives (RTPO8f)

### PR DESCRIPTION
**Pull request by sacOO7**
- Fixes https://ably.atlassian.net/browse/AIT-1138

## What

`PathObject#instance` previously returned undefined/null when the path resolved to a primitive value (`RTPO8d`). This PR deprecates that clause and replaces it with `RTPO8f`: a primitive resolution now returns a new `Instance` wrapping the primitive value. The corresponding UTS test spec is updated to match.

## Why

The primitive-null behaviour was the odd one out in the `Instance` model:

- `RTINS1` already defines `Instance` as "a direct-reference view of a `LiveObject` **or primitive value**".
- Primitives already surface as `Instance`s through `Instance#get` (`RTINS5c`), so a map entry read via an `Instance` produced a primitive `Instance`, while the same entry read via `PathObject#instance` produced nothing.
- `RTINS3b` (primitives have no object id) and `RTINS4c` (`Instance#value` returns the primitive directly) already specify how a primitive-wrapping `Instance` behaves — `RTPO8d` was just blocking the one entry point.

Making `instance()` total over resolved values gives callers a single, uniform way to capture a point-in-time reference to whatever sits at a path.

### `specifications/objects-features.md`

- `RTPO8d` — marked as replaced by `RTPO8f`, following the existing deprecation convention.
- `RTPO8f` (new) — a primitive resolution returns a new `Instance` (`RTINS1`) wrapping the primitive value.
- `RTPO8c` (LiveObject → `Instance`) and `RTPO8e` (resolution failure → undefined/null) are unchanged.

### `uts/objects/unit/path_object.md`

- `objects/unit/RTPO8c/instance-primitive-null-0` becomes `objects/unit/RTPO8f/instance-primitive-wrapped-0`: asserts the returned `Instance` exists, has no object id (`RTINS3b`), and exposes the primitive via `value()` (`RTINS4c`).

## SDK adoption

Both SDK implementations and their derived UTS tests have been updated to this behaviour and are pending on their respective branches:

- **ably-js**: `_resolveInstance` wraps any resolved value; `batch()` retains its explicit LiveObject check so the 92007 non-LiveObject error is unaffected.
- **ably-java**: `DefaultPathObject.instance()` returns the typed `Instance` for any resolved value; primitive instances remain anonymous (no `id` member) per the typed-SDK contract.

## Follow-up (separate PR)

The typed-SDK spec section (`RTTS6e` / `RTTS3b`, #491) still states that `instance()` on primitive `PathObject` sub-classes returns null or throws 92007 — that premise is removed by this change and will be reconciled in an individual PR against that branch.

### CODE REVIEW

**Comment by paddybyers** [comment]
Does this need a uts test update?

**Comment by sacOO7** [comment]
> Does this need a uts test update?

Yes, we need to update current test `instance() returns null for primitive` as per updated spec point.

**Review by ttypic — APPROVED** [review]
LGTM

File: specifications/objects-features.md [hunk 1]
```diff
@@ -912,8 +912,9 @@ A `PathObject` is obtained from `RealtimeObject#get` ([RTO23](#RTO23)), which re
   - `(RTPO8a)` Checks the access API preconditions per [RTO25](#RTO25)
   - `(RTPO8b)` Resolves the path using the path resolution procedure ([RTPO3](#RTPO3))
   - `(RTPO8c)` If the resolved value is a `LiveObject` (i.e. a `LiveMap` or `LiveCounter`), returns a new `Instance` ([RTINS1](#RTINS1)) wrapping that `LiveObject`
-  - `(RTPO8d)` If the resolved value is a primitive, returns undefined/null
+  - `(RTPO8d)` This clause has been replaced by [RTPO8f](#RTPO8f).
   - `(RTPO8e)` If path resolution fails, returns undefined/null per [RTPO3c1](#RTPO3c1)
+  - `(RTPO8f)` If the resolved value is a primitive, returns a new `Instance` ([RTINS1](#RTINS1)) wrapping the primitive value
```
[the remark this answers is not in the document]
Comment (sacOO7) [reply specifications/objects-features.md]: Addressed in 47382d8f: `RTPO8f` now enumerates the primitive types (`Boolean`, `Binary`, `Number`, `String`, `JsonArray`, `JsonObject`), matching `RTPO7d` / `RTINS4c`, and the clause has been moved above `RTPO8e` so the failure case is listed last, consistent with `RTPO7` and `RTPO13`. The ID itself is unchanged, since per CONTRIBUTING.md new IDs must be greater than all existing ones in the section.
